pkg/controller/directvolumemigrationprogress/directvolumemigrationprogress_controller_test.go

```diff
@@ -336,3 +340,565 @@ func Test_getLastMatch(t *testing.T) {
+						Conditions: migapi.Conditions{
+							List: []migapi.Condition{
+								{
+									Type:     "Ready",
+									Status:   "True",
+									Category: "Required",
+								},
+							},
+						},
+					},
+				},
+			},
+			args: args{
+				podRef: &kapi.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "dvm-rsync",
+						Namespace:         "test-app-1",
+						CreationTimestamp: metav1.Time{},
+					},
+					Spec: kapi.PodSpec{
+						Containers: []kapi.Container{
+							{
+								Name: "stunnel",
+							},
+							{
+								Name: "rsync-client",
+							},
+						},
+					},
+					Status: kapi.PodStatus{
+						ContainerStatuses: []kapi.ContainerStatus{
+							{
+								Name: "stunnel",
+								State: kapi.ContainerState{
+									Running: &kapi.ContainerStateRunning{
+										StartedAt: metav1.Time{},
+									},
+								},
+							},
+							{
+								Name:  "rsync-client",
+								Ready: false,
+								LastTerminationState: kapi.ContainerState{
+									Terminated: &kapi.ContainerStateTerminated{
+										FinishedAt: metav1.Time{},
+										ExitCode:   0,
+									},
+								},
+							},
+						},
+					},
+				},
+				p: fakeGetPodLogs{
+					podLogMessage: "",
+					err:           nil,
+				},
+			},
+			want: &migapi.RsyncPodStatus{
+				PodName:                     "dvm-rsync",
+				PodPhase:                    kapi.PodSucceeded,
+				LogMessage:                  "",
+				CreationTimestamp:           &metav1.Time{},
+				LastObservedProgressPercent: "100%",
+				LastObservedTransferRate:    "",
+				ExitCode:                    &zero,
+			},
+			wantErr: false,
+		},
+		{
+			name: "Given a succeeded Pod, appropriate container status is returned",
+			fields: fields{
+				Client:    fake.NewFakeClient(),
+				SrcClient: fakecompat.NewFakeClient(),
+				Cluster: &migapi.MigCluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "migcluster-host",
+						Namespace: "openshift-migration",
+					},
+					Spec: migapi.MigClusterSpec{
+						IsHostCluster: false,
+						Insecure:      true,
+					},
+					Status: migapi.MigClusterStatus{
+						Conditions: migapi.Conditions{
+							List: []migapi.Condition{
+								{
+									Type:     "Ready",
+									Status:   "True",
+									Category: "Required",
+								},
+							},
+						},
+					},
+				},
+			},
+			args: args{
+				podRef: &kapi.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "dvm-rsync",
+						Namespace:         "test-app-1",
+						CreationTimestamp: metav1.Time{},
+					},
+					Spec: kapi.PodSpec{
+						Containers: []kapi.Container{
+							{
+								Name: "stunnel",
+							},
+							{
+								Name: "rsync-client",
+							},
+						},
+					},
+					Status: kapi.PodStatus{
+						Phase: kapi.PodSucceeded,
+						ContainerStatuses: []kapi.ContainerStatus{
+							{
+								Name: "stunnel",
+								State: kapi.ContainerState{
+									Running: &kapi.ContainerStateRunning{
```

Comment:
if the pod succeeded this should not be running right? it should also be terminated?

Reply:
Done, updated the PR